gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/proc/LaunchDagProc.java

```diff
@@ -0,0 +1,205 @@
+import org.apache.gobblin.service.modules.orchestration.task.LaunchDagTask;
+import org.apache.gobblin.service.modules.spec.JobExecutionPlan;
+
+
+/**
+ * An implementation of {@link DagProc} for launching {@link org.apache.gobblin.service.modules.orchestration.task.DagTask}.
+ */
+@Slf4j
+@Alpha
+public class LaunchDagProc extends DagProc<Dag<JobExecutionPlan>, Void> {
+  private final LaunchDagTask launchDagTask;
+  NewDagManager newDagManager;
+
+  public LaunchDagProc(LaunchDagTask launchDagTask, NewDagManager newDagManager) {
+    this.launchDagTask = launchDagTask;
+    this.newDagManager = newDagManager;
+  }
+
+  @Override
+  protected Dag<JobExecutionPlan> initialize(DagManagementStateStore dagManagementStateStore) throws IOException {
+    return dagManagementStateStore.getDag(this.launchDagTask.getDagId().toString());
+  }
+
+  @Override
+  protected Void act(DagManagementStateStore dagManagementStateStore, Dag<JobExecutionPlan> dag) throws IOException {
+    if (dag == null) {
+      log.warn("No dag with id " + this.launchDagTask.getDagId() + " found to launch");
+      return null;
+    }
+    initializeDag(dag, dagManagementStateStore);
+    DagManagerUtils.submitPendingExecStatus(dag, this.eventSubmitter);
+    return null;
+  }
+
+  protected void initializeDag(Dag<JobExecutionPlan> dag, DagManagementStateStore dagManagementStateStore)
+      throws IOException {
+    //Add Dag to the map of running dags
+    String dagId = DagManagerUtils.generateDagId(dag).toString();
+    log.info("Initializing Dag {}", DagManagerUtils.getFullyQualifiedDagName(dag));
+
+    //A flag to indicate if the flow is already running.
+    boolean isDagRunning = false;
+    //Are there any jobs already in the running state? This check is for Dags already running
+    //before a leadership change occurs.
+    for (Dag.DagNode<JobExecutionPlan> dagNode : dag.getNodes()) {
+      if (DagManagerUtils.getExecutionStatus(dagNode) == ExecutionStatus.RUNNING) {
+        dagManagementStateStore.addJobState(dagId, dagNode);
+        //Update the running jobs counter.
+        NewDagManager.getDagManagerMetrics().incrementRunningJobMetrics(dagNode);
+        isDagRunning = true;
+      }
+    }
+
+    FlowId flowId = DagManagerUtils.getFlowId(dag);
+    NewDagManager.getDagManagerMetrics().registerFlowMetric(flowId, dag);
+
+    log.debug("Dag {} submitting jobs ready for execution.", DagManagerUtils.getFullyQualifiedDagName(dag));
+    //Determine the next set of jobs to run and submit them for execution
+    Map<String, Set<Dag.DagNode<JobExecutionPlan>>> nextSubmitted = submitNext(dagManagementStateStore, dagId);
+    for (Dag.DagNode<JobExecutionPlan> dagNode : nextSubmitted.get(dagId)) {
+      dagManagementStateStore.addJobState(dagId, dagNode);
+    }
+
+    // Set flow status to running
+    DagManagerUtils.emitFlowEvent(this.eventSubmitter, dag, TimingEvent.FlowTimings.FLOW_RUNNING);
```

Comment:
I originally though `act` might set things up`, but `commit` would actually record them and therefore send out GTEs of this sort... we should discuss...